### PR TITLE
calculateExtensionScriptUrl should return fully qualified Urls

### PR DESCRIPTION
**Pull request by aghassemi**
Fixed https://github.com/ampproject/amphtml/issues/4616

### CODE REVIEW

**Comment by lannka** [comment]
LGTM

**Comment by tdrl** [comment]
I was fighting fires yesterday, so I didn't have a chance to comment, but:  __THANK YOU!___  I tried to use fully qualified paths in the first version of this, but a reviewer recommended relative paths instead.  Only after it was rolled out did I realize that that breaks when using local dev mode.  :-P  Been meaning to fix this for a while, but...  Fires.  Thanks for fixing it!

File: src/service/extensions-impl.js [hunk 1]
```diff
@@ -440,8 +440,9 @@ export class Extensions {
     scriptElement.setAttribute('custom-element', extensionId);
     scriptElement.setAttribute('data-script', extensionId);
     const pathStr = this.win.location.pathname;
+    const base = this.win.location.protocol + '//' + this.win.location.host;
```
Comment (jridgewell) [inline src/service/extensions-impl.js]: We'll need port number, too. How about passing in `window.location`, and let calculateExtensionScriptUrl figure out what it needs.

File: src/service/extensions-impl.js [hunk 2]
```diff
@@ -452,23 +453,24 @@ export class Extensions {
...
         || path.indexOf('.max') >= 0 || path.substr(0, 5) == '/max/') {
-      return `/dist/v0/${extensionId}-0.1.max.js`;
+      return `${base}/dist/v0/${extensionId}-0.1.max.js`;
     }
     if ((isTest && isUsingCompiledJs)
         || path.indexOf('.min') >= 0 || path.substr(0, 5) == '/min/') {
-      return `/dist/v0/${extensionId}-0.1.js`;
+      return `${base}/dist/v0/${extensionId}-0.1.js`;
     }
     return `https://cdn.ampproject.org/v0/${extensionId}-0.1.js`;
```
Comment (jridgewell) [inline src/service/extensions-impl.js]: `${urls.cdn}/v0/...`

File: src/service/extensions-impl.js (hunk 1 above)
Comment (aghassemi) [inline src/service/extensions-impl.js]: `host` has the port number (`hostname` doesn't). Will send a PR addressing the two other comments soon.

Comment (aghassemi) [inline src/service/extensions-impl.js]: Looks like you did some nice refactoring in your SW PR. Thanks!

**Pull request merged**